vars/retrievePreviousBuild.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+/** Library to call build.sh script (builds OpenSearch and OpenSearch Dashboards distribution)
+ @param Map args = [:] args A map of the following parameters
+ @param args.inputManifest <required> - Relative path to input manifest containing all the components to build.
+ @param args.platform <required> - Platform of previous build to retrieve.
+ @param args.architecture <required> - Architecture of previous build to retrieve.
+ @param args.distribution <required> - Distribution of previous build to retrieve.
+ @param args.incremental <required> - String value contains the build number of previous build or the latest.
+ */
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@jenkins_incremental', retriever: legacySCM(scm))
+    def inputManifestObj = lib.jenkins.InputManifest.new(readYaml(file: args.inputManifest))
+
+    def DISTRIBUTION_JOB_NAME = "${JOB_NAME}"
+    def revision = inputManifestObj.build.version
+
+    def DISTRIBUTION_PLATFORM = args.platform
+    def DISTRIBUTION_ARCHITECTURE = args.architecture
+    def distribution = args.distribution
+    def prefixPath = "${WORKSPACE}/download"
+    def DISTRIBUTION_BUILD_NUMBER
+
+    if (args.incremental.equalsIgnoreCase("latest")) {
+        withCredentials([string(credentialsId: 'jenkins-artifact-bucket-name', variable: 'ARTIFACT_BUCKET_NAME')]) {
+            downloadFromS3(
+                    assumedRoleName: "opensearch-bundle",
+                    roleAccountNumberCred: "jenkins-aws-account-public",
+                    downloadPath: "${DISTRIBUTION_JOB_NAME}/${revision}/index.json",
+                    bucketName: "${ARTIFACT_BUCKET_NAME}",
+                    localPath: "${prefixPath}/index.json",
+                    force: true,
+            )
+        }
```

Comment:
I didnt realize this is just a index.json, then curl with `|` would do the trick instantly.
I would agree if it is the full ledge distribution folder, s3 is sufficient.
But for a single file and you only need to parse the value, wouldnt http is easier to go?

Reply:
Yes that's true. However, the URL for downloading the `index.json` from beta jenkins for testing is not available. That's the reason I used s3Download to be compatible for both.

Reply:
I think we can change to curl or wget. Credentials should not be required if the file path is known.

Reply:
Oh right! Downloading directly from S3 would cause issues. Better to use curl/wget to get index.json. @Divyaasm Can add more.
Thanks!

Reply:
Just sync up with @Divyaasm, we will be having different paths for index.json and it's unique for different type of distribution/architecture/platform. 

For now, I can use `curl` to download from current location for index.json 
e.g. `https://ci.opensearch.org/ci/dbc/${DISTRIBUTION_JOB_NAME}/${revision}/index.json`

But in the future once we have new `index.json` for each distribution/architecture/platform type, we would need to update the URL here to 
`https://ci.opensearch.org/ci/dbc/${DISTRIBUTION_JOB_NAME}/${revision}/index/${platform}/${architecture}/${distribution}/index.json`

What do you think? @gaiksaya @peterzhuamazon

Reply:
I think it should be always curl. downloadfromS3 is over complicating a single file download.

Reply:
Sure. I have to use s3download since we don't have that URL available for beta Jenkins. But in here, we can use curl

Reply:
Updated in the latest commit. Please help take a look. @gaiksaya @peterzhuamazon